src/main/java/org/springframework/data/redis/connection/RedisNode.java

```diff
@@ -121,11 +182,21 @@ public RedisNodeBuilder() {
 			node = new RedisNode();
 		}
 
+		/**
+		 * Define node name.
+		 */
 		public RedisNodeBuilder withName(String name) {
 			node.name = name;
 			return this;
 		}
 
+		/**
+		 * Set host and port of server.
+		 * 
+		 * @param host must not be {@literal null}.
+		 * @param port
+		 * @return
+		 */
 		public RedisNodeBuilder listeningAt(String host, int port) {
 
 			Assert.hasText(host, "Hostname must not be empty or null.");
```

Comment:
`host` can be empty (`null` RedisURI in lettuce) in a cluster when the node has the `noaddr` flag